coderd/database/migrations/000338_workspace_app_status_idle.down.sql

```diff
@@ -0,0 +1 @@
+-- It is not possible to delete enum values.
```

Comment:
I don't think that's true. These steps should revert the enum to its original set of values:
1. Replace any occurences of new enum values with different values
2. Rename the enum to something else
3. Re-create the original enum
4. Update columns to use the new enum
5. Drop the renamed enum

Here are the SQL statements you'd use: https://blink.so/chat/Revert-Enum-Value-in-Postgres-5vVDbfY5A7ZOLPxyuqVgsG

Reply:
Ah yeah I meant there is no `ALTER DELETE` to mirror `ALTER ADD` but yeah we could recreate the entire enum.  But from what I saw this is how our other migrations handle enum downgrades.

I am not sure if that means it is a good idea, but it is consistent at least.

Although...I wonder if this causes issues with queries or any enum validation in the Go code.  Maybe we need to add downgrades to all of these enums.  :cry:

Reply:
I added the down migration after our discussion yesterday.

Actually, after a bit more searching I found that we *do* have some precedence, for example: https://github.com/coder/coder/blob/main/coderd/database/migrations/000336_add_organization_port_sharing_level.down.sql#L5-L10